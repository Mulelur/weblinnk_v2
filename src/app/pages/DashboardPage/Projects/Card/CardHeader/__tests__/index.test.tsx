import * as React from 'react';
import { Store } from '@reduxjs/toolkit';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styles/theme/ThemeProvider';
import { HelmetProvider } from 'react-helmet-async';
import { configureAppStore } from 'store/configureStore';
import { CardHeader } from '..';

const renderCard = (store: Store) =>
  render(
    <Provider store={store}>
      <ThemeProvider>
        <HelmetProvider>
          <CardHeader siteName="New Site" />
        </HelmetProvider>
      </ThemeProvider>
    </Provider>,
  );

describe('<CardHeader />', () => {
  let component: ReturnType<typeof renderCard>;

  beforeEach(() => {
    const store = configureAppStore();
    component = renderCard(store);
  });

  it('should match snapshot', () => {
    expect(component.container).toMatchSnapshot();
  });

  it('should render the title', () => {
    expect(component.getByText('Projects')).toBeInTheDocument();
  });
});
