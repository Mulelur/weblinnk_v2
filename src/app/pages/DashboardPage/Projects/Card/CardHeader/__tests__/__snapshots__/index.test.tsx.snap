// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardHeader /> should match snapshot 1`] = `
.c0 {
  width: 100%;
  height: 3.2rem;
  border-radius: 0.25rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <div
    class="c0"
  >
    Projects
  </div>
</div>
`;
